code/Model/SiteTree.php

```diff
@@ -2711,6 +2712,43 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @param string $action
+	 * @return Controller
+	 */
+	public function getAssociatedController($action = null) {
+		$controllerName = $this->getControllerName($action);
+		return Injector::inst()->create($controllerName, $this);
+	}
+
+	/**
+	 * @param string $action
+	 * @return string
+	 */
+	public function getControllerName($action = null) {
+		$action = ucfirst($action); // Needed as Injector service names are case sensitive
```

Comment:
hmm, this is why injector needs to be case insensitive, class names are and this shouldn't be something we do - it feels risky... but if that's how it was done before... carry on